### PR TITLE
Typos fixed, QOL updates.

### PR DESCRIPTION
**Pull request by WizardCell**
I spent the last two months reading through GASDocumentation and it was a fun journey. While I was reading, I made sure to notate any typos I found along the way. I barely could find any compared to how monolithic the doc is. Hopefully this PR has covered all the typos!

### CODE REVIEW

**Comment by WizardCell** [comment]
Some of the fixes I did was that I replaced some of the of UGD prefixes (which stand for GASDocumentation) to UGS (which stand for (GAS Shooter, i.e. the Sample Project). Maybe I shouldn't have done that, but if you look closely that caused me some confusion as in this case for example:

image


Before it used to say "UGD", and me as a reader was confused to see that the Sample Project uses some other totally different prefix. Let me know what you think regarding that? (I'm asking because in my PR I didn't replace all "GD" prefixes to "GS", which I noticed quite late).

For example, in this case:

image

I didn't replace UGD to UGS because this code doesn't exist at all the GAS Shooter Sample Project.

**Review by tranek — CHANGES_REQUESTED** [review]
The prefixes are correct and don't need to be changed. Please remove those and I'll look at the other typos.

**Comment by tranek** [comment]
Hello,

Thanks for looking into typos. The `Sample Project` generally refers to the Unreal project included in the GASDocumentation repository https://github.com/tranek/GASDocumentation/tree/master/Source/GASDocumentation. The prefixes for the code snippets should be correct. I understand why it's confusing since the snippets were pulled from many different projects - GD, GS, and even a few PA (my ParagonAssets project that's not public).

File: README.md [hunk 1]
```diff
@@ -1852,10 +1852,10 @@ We grant `GameplayAbilities` on the server which then automatically replicates t
 
 The Sample Project stores a `TArray<TSubclassOf<UGDGameplayAbility>>` on the `Character` class that it reads from and grants when the game starts:
 ```c++
-void AGDCharacterBase::AddCharacterAbilities()
+void AGSCharacterBase::AddCharacterAbilities()
```
Comment (tranek) [inline README.md:1855]: Should still be AGDCharacterBase

File: README.md [hunk 2]
```diff
@@ -1852,10 +1852,10 @@ We grant `GameplayAbilities` on the server which then automatically replicates t
...
 {
 	// Grant abilities, but only on the server	
-	if (Role != ROLE_Authority || !AbilitySystemComponent.IsValid() || AbilitySystemComponent->CharacterAbilitiesGiven)
+	if (Role != ROLE_Authority || !AbilitySystemComponent.IsValid() || AbilitySystemComponent->bCharacterAbilitiesGiven)
```
Comment (tranek) [inline README.md:1858]: If we change it here, we should change it in the sample project code too.

File: README.md [hunk 3]
```diff
@@ -1866,7 +1866,7 @@ void AGDCharacterBase::AddCharacterAbilities()
 			FGameplayAbilitySpec(StartupAbility, GetAbilityLevel(StartupAbility.GetDefaultObject()->AbilityID), static_cast<int32>(StartupAbility.GetDefaultObject()->AbilityInputID), this));
 	}
 
-	AbilitySystemComponent->CharacterAbilitiesGiven = true;
+	AbilitySystemComponent->bCharacterAbilitiesGiven = true;
```
Comment (tranek) [inline README.md:1869]: If we change it here, we should change it in the sample project code too.

File: README.md [hunk 4]
```diff
@@ -1933,9 +1933,9 @@ void UGDGameplayAbility::OnAvatarSet(const FGameplayAbilityActorInfo * ActorInfo
 {
 	Super::OnAvatarSet(ActorInfo, Spec);
 
-	if (ActivateAbilityOnGranted)
+	if (bActivateAbilityOnGranted)
```
Comment (tranek) [inline README.md:1936]: If we change it here, we should change it in the sample project code too.

File: README.md (hunk 1 above)
Comment (WizardCell) [reply README.md]: Sorry I missed this one, should be fixed in the last commit.

File: README.md (hunk 2 above)
Comment (WizardCell) [reply README.md:1858]: Unless I'm missing something, it's written that way in the sample project code (checked master and 5.1 branches).

File: README.md (hunk 3 above)
Comment (WizardCell) [reply README.md:1869]: Unless I'm missing something, it's written that way in the sample project code (checked master and 5.1 branches).

File: README.md (hunk 4 above)
Comment (WizardCell) [reply README.md:1936]: I would like to note that the next line, i.e. the line of the unused local variable was changed/fixed in the first revert commit, like you did in the 5.1 branch. Let me know if that's okay 😄

File: README.md (hunk 2 above)
Comment (tranek) [reply README.md:1858]: https://github.com/tranek/GASDocumentation/blob/master/Source/GASDocumentation/Private/Characters/GDCharacterBase.cpp#L277

File: README.md (hunk 3 above)
Comment (tranek) [reply README.md:1869]: https://github.com/tranek/GASDocumentation/blob/master/Source/GASDocumentation/Private/Characters/GDCharacterBase.cpp#L288

File: README.md (hunk 2 above)
Comment (WizardCell) [reply README.md:1858]: Ah I missed this source code. I was referring to the GASShooter corresponding code. This also explains what you meant at the start with there is no issue with the prefix (which I didn't fully understand/realize you had real code for GASDocumentation but just thought you meant the corresponding code in GAShooter with the prefix change).

I did a PR: https://github.com/tranek/GASDocumentation/pull/95
Sorry for the inconvenience!

File: README.md (hunk 4 above)
Comment (WizardCell) [reply README.md:1936]: Ah fair enough, that's also in the sample code which I missed. I fixed that in the new PR: https://github.com/tranek/GASDocumentation/pull/95

**Comment by tranek** [comment]
Thank you for doing this! 🙌

**Comment by WizardCell** [comment]
Actually, thank you Dan for bearing with me through the process! 😄

**Pull request merged**